nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -99,7 +101,11 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                 }
                 new WorkHolder(work).run();
                 if (storeState) {
-                    WorkStateHelper.setState(work.getId(), null, stateTTL);
+                    // if another work with same id exists we set the state to SCHEDULED, null otherwise
+                    Work.State state = WorkStateHelper.getLastOffset(work.getId()) > context.getLastOffset().offset()
```

Comment:
why are you testing the offset instead of the work state ? 
you could retrieve the state of the work and set it to null only if it is running, 
in both case it is a roundtrip to the kvstore and you don't have to mix different things